File: backend/java/localpos-user-service/src/main/java/com/pds/localpos/userservice/repository/UserRepository.java
package com.pds.localpos.userservice.repository;

import com.pds.localpos.userservice.model.User;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<User, String> {

    User findByUsername(String username);

    boolean existsByUsername(String username);

    boolean existsByEmail(String email);
}

File: backend/kotlin/localpos-user-service/src/main/kotlin/com/pds/localpos/userservice/repository/UserRepository.kt
package com.pds.localpos.userservice.repository

import com.pds.localpos.userservice.model.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserRepository : JpaRepository<User, Long> {

    fun findByUsername(username: String): Optional<User>
    fun existsByUsername(username: String): Boolean
    fun existsByEmail(email: String): Boolean
}
